src/include/cart/api.h

```diff
@@ -1947,6 +1947,65 @@ int crt_group_secondary_create(crt_group_id_t grp_name,
  */
 int crt_group_secondary_destroy(crt_group_t *grp);
 
+
+/**
+ * Perform a primary group modification in an atomic fashion based on the
+ * operation specified. Currently supported operations are 'add', 'remove'
+ * and 'replace'. This API allows multiple ranks to be added or removed
+ * at the same time with a single call.
+ *
+ * Add: Ranks in the rank list are added to the group with corresponding uris
+ * Remove: Ranks in the rank list are removed from the group
+ * Replace:
+ *     Ranks that exist in group and not in rank list get removed
+ *     Ranks that exist in rank list and not in group get added
+ *     Ranks that exist in both rank list and group are left unmodified
+ *
+ * \param[in] grp                Group handle
+ * \param[in] ctxs               Array of contexts
+ * \param[in] num_ctxs           Number of contexts
+ * \param[in] ranks              Modification rank list
+ * \param[in] uris               Array of URIs corresponding to contexts and
+ *                               rank list
+ * \param[in] op                 Modification operation.
+ *
+ * \return                       DER_SUCCESS on success, negative value on
+ *                               failure.
+ *
+ * Note: \ref uris is an array of srings; expected to be of size
+ * ranks->rl_nr * num_ctxs
```

Comment:
s/srings/strings/. How about: "\ref uris shall be an array of (ranks->rl_nr * num_ctxs) strings."? Related, if ctxs is

    ctxs[0]: of provider0
    ctxs[1]: of provider1

then do uris need to have provider0 URIs before provider1 URIs?